### PR TITLE
Create counting_sort.py

### PR DESCRIPTION
**Pull request by quang2705**

### CODE REVIEW

File: counting_sort.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+  def couting_sort(arr):
```
Comment (ankit167) [inline counting_sort.py]: typo- couting_sort. Same goes for line 23

File: counting_sort.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	result_arr = [0]*len(arr)
+	#creating a result_arr an put the element in a correct positon
+	for i in range(len(arr)-1,-1,-1):
+		result_arr[temp_arr[arr[i]]-1] = arr[i]-different
+		temp_arr[arr[i]] = temp_arr[arr[i]]-1
+	
+	return result_arr 
+
+
+x =counting_sort([1,2,3,4,9,1,2,8,3,5,7,0,9,8,1,7,4,5])
```
Comment (keon) [inline counting_sort.py]: can you wrap this with __main__?

**Comment by keon** [comment]
@quang2705 looks nice.
One last thing.
Can you convert it to 4 spaces instead of tabs?

**Comment by keon** [comment]
@quang2705 doesn't look like you converted all tabs to spaces.

**Comment by quang2705** [comment]
I have no idea what happened, it kept changing back to tab after I edited

**Comment by carlwright4** [comment]
Just started kinda lost

File: counting_sort.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+def counting_sort(arr):
+	"""
+    Couting_sort
```
Comment (ankit167) [inline counting_sort.py]: Typo- Couting_sort

File: counting_sort.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	different = 0
+	if m < 0:
+		#save the change, so that we can convert the array back to all positive number
+		different = -m
+		for i in range (len(arr)):
+			arr[i]+= -m
+	k = max(arr)
+	temp_arr = [0]*(k+1)
+	for i in range(0,len(arr)):
+		temp_arr[arr[i]]= temp_arr[arr[i]]+1
```
Comment (ankit167) [inline counting_sort.py]: Please use spaces around operators. Better to test syntax style with pep8 before pushing code.

**Pull request merged**